jung-samples/src/main/java/edu/uci/ics/jung/samples/BalloonLayoutDemo.java

```diff
@@ -89,8 +90,9 @@ public BalloonLayoutDemo() {
 
     vv.setGraphMouse(graphMouse);
     vv.addKeyListener(graphMouse.getModeKeyListener());
-
-    Lens lens = new Lens(vv);
+    LayoutModel layoutModel = vv.getModel().getLayoutModel();
```

Comment:
Consider giving VisualizationModel a method to return a Dimension based on the LayoutModel that it contains.  That way you don't have to do this two-line dance everywhere, you can just do this:

Lens lens = new Lens(vv.getModel().getDimension());

Downside is that you tie VisualizationModel a bit more to AWT, but it already has a setLayoutSize(Dimension) anyway.

Alternatively, you could have a static utility method that did this, but it seems reasonable to have an easy way to ask the model for its dimensions.

(Microscopic side benefit: you don't have to import LayoutModel, because you never need one.)

Reply:
Separation of concerns. The VisualizationModel does not have a Dimension and the setLayoutSize is a leftover artifact. I'll remove it. I meant to clean up stuff like that. Thanks for spotting it.

Reply:
I can understand wanting to keep Visualization Model AWT-free.  Considering the number of times you're doing that two-line dance, though, I'd still consider adding a static utility.

Do you plan to have a DimensionModel to go with our PointModel?

Reply:
I'll give that some thought (static method). It's not as bad as the vv.getRenderer().getBlah().getFoo(). etc instead of just plumbing in a vv.setTheColorOfTheNodeTextWhenItsHighlighted(Color)

Reply:
mostly, its keeping the policy that there are 2 main sizes: one for bounding the layoutalgorithm and one for the visualization window. I got rid of the VisualizationModel one because it was one too many places for the layoutModel size.